### PR TITLE
feat: change unit switch logic

### PR DESCRIPTION
**Pull request by limpbrains**
### Description

To simplify logic of choosing the format of displaying amounts, I've merged settings.bitcoinUnit and settings.balanceUnit into new settings.unit option. Also I've removed showFiat from <Money /> component and replaced it with unitType = primary | secondary. It is primary by default.

Primary is a unit from settings.unit, secondary is "opposite". This allows to easily show some amount in bitcoin / fiat, without bothering which is which, just <Money sats={100} size="big" /> and <Money sats={100} size="small" />

Also I've added primaryUnitSelector and secondaryUnitSelector to reflect these changes.

Also I've added ability to change unit by tapping on Amount for a few more screens.

### Type of change

Refactoring

### Tests

No test

### Screenshot / Video

https://github.com/synonymdev/bitkit/assets/155891/2e9ff397-83a9-4790-a69d-6df04343b815

### QA Notes

Make sure that changing unit works fine across all screens.

### CODE REVIEW

File: src/screens/Lightning/QuickSetup.tsx [hunk 1]
```diff
@@ -243,9 +255,7 @@ const QuickSetup = ({
 					<NumberPadTextField
 						value={textFieldValue}
 						showPlaceholder={showNumberPad}
-						reverse={true}
```
Comment (pwltr) [inline src/screens/Lightning/QuickSetup.tsx:246]: `reverse={true}` still needed

File: src/store/types/wallet.ts [hunk 2]
```diff
@@ -29,11 +29,6 @@ export type TAssetType = 'bitcoin' | 'tether';
 export enum EBitcoinUnit {
 	satoshi = 'satoshi',
 	BTC = 'BTC',
-}
-
-export enum EBalanceUnit {
-	satoshi = 'satoshi',
-	BTC = 'BTC',
```
Comment (pwltr) [inline src/store/types/wallet.ts]: Can you please change the variable name for the enum? I find it confusing that `EBitcoinUnit` contains a member 'fiat'. Also assignments like `bitcoinUnit` etc. Sorry nitpicking but helps with readability.

File: src/screens/Wallets/Send/Amount.tsx [hunk 3]
```diff
@@ -120,6 +119,8 @@ const Amount = ({ navigation }: SendScreenProps<'Amount'>): ReactElement => {
 
 	const onChangeUnit = (): void => {
 		const result = getNumberPadText(amount, nextUnit);
+		console.info('nextUnit', nextUnit);
+		console.info('result', result);
```
Comment (pwltr) [inline src/screens/Wallets/Send/Amount.tsx]: debug leftover

File: src/utils/displayValues/index.ts [hunk 4]
```diff
@@ -16,11 +16,15 @@ export const getBitcoinDisplayValues = ({
...
 }: {
 	satoshis: number;
-	bitcoinUnit?: EBitcoinUnit;
+	bitcoinUnit: EBitcoinUnit;
 }): IBitcoinDisplayValues => {
 	try {
 		if (!bitcoinUnit) {
-			bitcoinUnit = getSettingsStore().bitcoinUnit;
+			bitcoinUnit = getSettingsStore().unit;
+		}
```
Comment (pwltr) [inline src/utils/displayValues/index.ts]: Can remove this block because `bitcoinUnit` is now required

File: src/screens/Wallets/Send/Amount.tsx [hunk 5]
```diff
@@ -103,7 +103,6 @@ const Amount = ({ navigation }: SendScreenProps<'Amount'>): ReactElement => {
 	]);
 
 	const availableAmountProps = {
-		...(unit !== EBalanceUnit.fiat ? { symbol: true } : { showFiat: true }),
```
Comment (pwltr) [inline src/screens/Wallets/Send/Amount.tsx:106]: Symbol missing for BTC & sats

File: src/screens/Transfer/Setup.tsx [hunk 6]
```diff
@@ -258,7 +258,6 @@ const Setup = ({ navigation }: TransferScreenProps<'Setup'>): ReactElement => {
 					<NumberPadTextField
 						value={textFieldValue}
 						showPlaceholder={showNumberPad}
-						reverse={true}
```
Comment (pwltr) [inline src/screens/Transfer/Setup.tsx:261]: revert

Comment (pwltr) [reply src/screens/Transfer/Setup.tsx:261]: and I think there should be a "Custom Amount" button here as well like in QuickSetup?

File: src/screens/Wallets/Send/ReviewAndSend.tsx [hunk 7]
```diff
@@ -579,12 +570,18 @@ const ReviewAndSend = ({
...
+							navigation.goBack();
+						}}>
+						<Money
+							sats={amount}
+							size="display"
+							style={styles.amountToggle}
+							symbol={true}
+							decimalLength="long"
+						/>
+					</TouchableOpacity>
```
Comment (pwltr) [inline src/screens/Wallets/Send/ReviewAndSend.tsx]: secondary (fiat/sats) is missing, also if coin selection is active then `onPress` should still navigate to Amount screen

File: src/screens/Activity/ActivityDetail.tsx [hunk 8]
```diff
@@ -347,11 +349,11 @@ const OnchainActivityDetail = ({
...
 				sats={value}
-				unit={EBalanceUnit.fiat}
 				size="caption13Up"
 				color="gray1"
+				unitType="secondary"
 			/>
-			<View style={styles.title}>
+			<TouchableOpacity style={styles.title} onPress={switchUnit}>
 				<View style={styles.titleBlock}>
 					<Money sats={value} sign={isSend ? '- ' : '+ '} />
```
Comment (pwltr) [inline src/screens/Activity/ActivityDetail.tsx:354]: Fiat has the symbol here but not BTC/sats, looks inconsistent

File: src/screens/Wallets/WalletsDetail/index.tsx [hunk 9]
```diff
@@ -191,13 +179,30 @@ const WalletsDetail = ({
...
+										entering={FadeIn}
+										exiting={FadeOut}>
+										<Money
+											sats={totalBalance}
+											enableHide={true}
+											highlight={true}
+											size="caption13M"
+											unitType="secondary"
+											color="gray1"
+										/>
```
Comment (pwltr) [inline src/screens/Wallets/WalletsDetail/index.tsx:195]: again, fiat has the symbol but not BTC/sats, maybe need to clarify with Aldert

**Review by pwltr — CHANGES_REQUESTED** [review]

File: src/store/types/wallet.ts (hunk 2 above)
Comment (limpbrains) [reply src/store/types/wallet.ts]: np, these are all valid concerns

File: src/screens/Activity/ActivityDetail.tsx (hunk 8 above)
Comment (limpbrains) [reply src/screens/Activity/ActivityDetail.tsx:354]: I've asked Alders, will fix it in another PR, if needed

File: src/screens/Wallets/WalletsDetail/index.tsx (hunk 9 above)
Comment (limpbrains) [reply src/screens/Wallets/WalletsDetail/index.tsx:195]: I've asked Alders, will fix it in another PR, if needed

**Comment by limpbrains** [comment]
@pwltr thx for awesome review! 
I think I've fixed all the problems, please take a look again

File: src/store/types/wallet.ts [hunk 10]
```diff
@@ -26,17 +26,19 @@ export type TKeyDerivationChange = '0' | '1'; //"Receiving Address" | "Change Ad
...
 	fiat = 'fiat',
 }
 
+export type TBitcoinLabel =
+	| 'Bitcoin Mainnet'
+	| 'Bitcoin Testnet'
+	| 'Bitcoin Regtest';
+
+export type TTicker = 'BTC' | 'tBTC';
+
```
Comment (pwltr) [inline src/store/types/wallet.ts]: this was merged back in, should be removed

File: src/utils/helpers.ts [hunk 11]
```diff
@@ -30,6 +32,46 @@ export const isOnline = async (): Promise<boolean> => {
...
+
+export const getLastWordInString = (phrase = ''): string => {
+	try {
+		const n = phrase.split(' ');
+		return n[n.length - 1];
+	} catch (e) {
+		return phrase;
+	}
+};
+
```
Comment (pwltr) [inline src/utils/helpers.ts]: also merged back in, should be removed

File: src/utils/helpers.ts [hunk 12]
```diff
@@ -164,19 +206,6 @@ export const capitalize = (text = ''): string => {
...
- */
-export const getLastWordInString = (phrase: string): string => {
-	try {
-		const n = phrase.split(' ');
-		return n[n.length - 1];
-	} catch (e) {
-		return phrase;
-	}
-};
-
```
Comment (pwltr) [inline src/utils/helpers.ts]: keep this

**Review by pwltr — CHANGES_REQUESTED** [review]

**Comment by pwltr** [comment]
Please squash commits

File: src/utils/helpers.ts (hunk 11 above)
Comment (limpbrains) [reply src/utils/helpers.ts]: fixed

File: src/utils/helpers.ts (hunk 12 above)
Comment (limpbrains) [reply src/utils/helpers.ts]: fixed

**Comment by limpbrains** [comment]
> Please squash commits

I will, after review is complete

**Review by pwltr — APPROVED** [review]

**Pull request merged**